keycloak/keycloak-admin-ui
# Makes jumplinks work

## Motivation
This connects up jumplinks and also pulls in a new Patternfly to get a fix to the JumpLinks component.

## Brief Description
This sets the scrollContainer so that the jumplinks component knows what to sync with. Currently, the component does not handle spaces in ids, so spaces are changed to hyphens. CSS changed to remove module css and use the BEM naming standard.

## Verification Steps
Jumplinks can be seen on the client settings page. Scroll the page and click the jump links.
Note that the highlight is based on what section is at the top of the page - so if the section clicked is far enough down that the page cannot scroll it to the top, the section highlighted will not be what was just clicked. However, user focus will be placed on the section that was selected.

## Checklist:

- [NA] Help has been implemented

## Additional Notes
Note that tabOrder is put onto the section headings in order for the jump link to throw focus to it. Although this is usually not a great idea since a heading is not an interactive element, this is deemed better for screen readers so that the next thing read is the section that you jumped to. It also prevents a bug showing two highlighted sections in some cases. https://github.com/patternfly/patternfly/issues/3747

Finally, I have no idea why one line in jsonFileUpload.test.txs.snap changed. It may be related to the PatternFly update??
